app/models/user/permissions.rb

```diff
@@ -79,13 +79,17 @@ def existing_service_ids
   # but then it is much harder to test MemberPermission#service_ids=
   # returns [] if no services are enabled, and nil if all (current and future) services are enabled
   def member_permission_service_ids
-    return nil if admin? || !services_member_permission
-    permitted_service_ids = services_member_permission.try(:service_ids) || []
-    permitted_service_ids & existing_service_ids
+    @member_permission_service_ids ||=
```

Comment:
if you want to memoize `nil`, you have to do like:
```
return @member_permission_service_ids id defined?(@member_permission_service_ids)

@member_permission_service_ids = ...
```

Reply:
Oh yes, true, I didn't I was memoizing `nil` :grimacing:  Thanks!